app/(main)/leaderboard/agents/AgentsLeaderboardTable.tsx

```diff
@@ -0,0 +1,233 @@
+          <TableRow>
+            <TableHead>Agent</TableHead>
+            <SortableHeader sortKey="profit_loss">Profit/Loss</SortableHeader>
+            <SortableHeader sortKey="total_volume">Volume Traded</SortableHeader>
+            <SortableHeader sortKey="success_rate">Success Rate</SortableHeader>
+            <SortableHeader sortKey="total_wins">Won bets</SortableHeader>
+            <SortableHeader sortKey="total_bets">Total bets</SortableHeader>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {agentsLeaderboardData.map(agent => (
+            <TableRow key={agent.address}>
+              <TableCell>
+                <div className="flex w-fit flex-shrink-0 items-center space-x-2 text-sm md:text-base">
+                  <Avatar address={agent.address} />
+                  <Link
+                    href={`/profile?address=${agent.address}`}
+                    className="hover:underline"
+                  >
+                    {agent.label}
+                  </Link>
+                </div>
+              </TableCell>
+              <TableCell
+                className={twMerge(
+                  'text-right',
+                  agent.profit_loss >= 0.01 && 'text-text-success-main',
+                  agent.profit_loss <= -0.01 && 'text-text-danger-main'
+                )}
+              >
+                {agent.profit_loss.toLocaleString('en-US', {
+                  style: 'currency',
+                  currency: 'USD',
+                })}
+              </TableCell>
+              <TableCell className="text-right">
+                {agent.total_volume.toLocaleString('en-US', {
+                  style: 'currency',
+                  currency: 'USD',
+                })}
+              </TableCell>
+              <TableCell className="text-right">{agent.success_rate}%</TableCell>
+              <TableCell className="text-right">
+                {agent.total_wins.toLocaleString()}
+              </TableCell>
+              <TableCell className="text-right">
+                {agent.total_bets.toLocaleString()}
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+      <PaginationControls
+        page={page}
+        totalPages={totalPages}
+        handlePageChange={handlePageChange}
+      />
+    </div>
+  );
+}
+
+const PaginationControls = ({
+  page,
+  totalPages,
+  handlePageChange,
+}: {
+  page: number;
+  totalPages: number;
+  handlePageChange: (newPage: number) => void;
+}) => {
+  const showFirstPage = page > 2;
+  const showLastPage = page < totalPages - 1;
+  const pageNumbers = [];
+
+  if (showFirstPage) {
+    pageNumbers.push(1);
+    if (page > 3) pageNumbers.push('...');
+  }
+
+  for (let i = Math.max(1, page - 1); i <= Math.min(totalPages, page + 1); i++) {
+    pageNumbers.push(i);
+  }
+
+  if (showLastPage) {
+    if (page < totalPages - 2) pageNumbers.push('...');
+    pageNumbers.push(totalPages);
+  }
```

Comment:
Shouldn't this be inside a `useEffect`? Otherwise it runs every time it renders

Reply:
The calculation needs to happen on every render to correctly show the pagination numbers as they change. 

When the user clicks to a different page, we want these calculations to run again to update the displayed page numbers.